cmd/client-s3.go

```diff
@@ -923,6 +925,23 @@ func isVirtualHostStyle(host string) bool {
 	return isAmazon(host) && !isAmazonChina(host) || isGoogle(host) || isAmazonAccelerated(host)
 }
 
+// getRegionFromHost - parse region from host name if present.
+func getRegionFromHost(host string) (region string) {
+	region = ""
+	if isGoogle(host) {
```

Comment:
You can use s3utils.IsGoogleCloudEndpoint() - perhaps combine below if and Google in a switch case and return?